src/commons/api/history.rs

```diff
@@ -515,85 +585,116 @@ impl fmt::Display for StorableCaCommand {
-            StorableCaCommand::UpdateRcvdCert(rcn, resources) => write!(
+            StorableCaCommand::UpdateRcvdCert {
+                resource_class_name,
+                resources,
+            } => write!(
                 f,
                 "Update received cert in RC '{}', with resources '{}'",
-                rcn,
+                resource_class_name,
                 resources.summary()
             ),
 
             // ------------------------------------------------------------
             // Key rolls
             // ------------------------------------------------------------
-            StorableCaCommand::KeyRollInitiate(duration) => {
-                write!(f, "Initiate key roll for keys older than '{}' seconds", duration)
+            StorableCaCommand::KeyRollInitiate { older_than_seconds } => {
+                write!(
+                    f,
+                    "Initiate key roll for keys older than '{}' seconds",
+                    older_than_seconds
+                )
             }
-            StorableCaCommand::KeyRollActivate(duration) => {
-                write!(f, "Activate new keys staging longer than '{}' seconds", duration)
+            StorableCaCommand::KeyRollActivate { staged_for_seconds } => {
+                write!(
+                    f,
+                    "Activate new keys staging longer than '{}' seconds",
+                    staged_for_seconds
+                )
             }
 
-            StorableCaCommand::KeyRollFinish(rcn) => write!(f, "Retire old revoked key in RC '{}'", rcn),
+            StorableCaCommand::KeyRollFinish { resource_class_name } => {
+                write!(f, "Retire old revoked key in RC '{}'", resource_class_name)
+            }
 
             // ------------------------------------------------------------
             // ROA Support
             // ------------------------------------------------------------
-            StorableCaCommand::RoaDefinitionUpdates(updates) => write!(
-                f,
-                "Update ROAs add: {} remove: '{}'",
-                updates.added().len(),
-                updates.removed().len()
-            ),
+            StorableCaCommand::RoaDefinitionUpdates { updates } => {
+                write!(f, "Update ROAs",)?;
+                if !updates.added().is_empty() {
+                    write!(f, "  ADD:",)?;
+                    for addition in updates.added() {
+                        write!(f, " {}", addition)?;
+                    }
+                }
+                if !updates.removed().is_empty() {
+                    write!(f, "  REMOVE:",)?;
+                    for rem in updates.removed() {
+                        write!(f, " {}", rem)?;
+                    }
+                }
+                Ok(())
+            }
+            StorableCaCommand::AutomaticRoaRenewal => {
+                write!(f, "Automatically re-issue ROAs before they would expire")
```

Comment:
I'm not sure from the issue descriptions how this relates to either #370 or #423, does it or is it new separate functionality?